File: Part8/C/FunctionConcepts/FunctionConcepts_4.c
// Pizza example with input parameters
// Prints the total price of a pizza based on size, toppings and delivery.
// Size_Par: a string as either small, medium or large.
// ToppingCount_Par: integer 0+
// DeliveryRequested_Par: Delivery has been requested if true.
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

const double TOPPING_PRICE = 1.50;
const double DELIVERY_FEE = 3.00;

void printPizzaCost ( char* Size_Par, int ToppingCount_Par, bool DeliveryRequested_Par )
{
    double basePrice = 0.0;
    
    if ( strcmp ( Size_Par, "small" ) == 0 )
    {
        basePrice = 12.99;
    }
    else if ( strcmp ( Size_Par, "medium" ) == 0 )
    {
        basePrice = 15.99;
    }
    else
    {
        basePrice = 18.99;
    }
    
    double toppingCost = ToppingCount_Par * TOPPING_PRICE;
    
    double deliveryFee = 0;
    if ( DeliveryRequested_Par == true )
    {
        deliveryFee = DELIVERY_FEE;
    }
    
    double totalPrice = basePrice + toppingCost + deliveryFee;
    
    printf ( "Pizza order:\n" );
    printf ( "\tBase price: %.2f\n", basePrice );
    printf ( "\tTopping price: %.2f\n", toppingCost );
    printf ( "\tDelivery price: %.2f\n", deliveryFee );
    printf ( "\t\tTotal price: %.2f\n", totalPrice );
}

int main ()
{
    printPizzaCost ( "small", 2, true );
    return 0;
}

File: Part8/Python/FunctionConcepts/FunctionConcepts_4.py
# Pizza example with input parameters
# Prints the total price of a pizza based on size, toppings and delivery.
# Size_Par: a string as either small, medium or large.
# ToppingCount_Par: integer 0+
# DeliveryRequested_Par: Delivery has been requested if true.

TOPPING_PRICE = 1.50
DELIVERY_FEE = 3.00

def printPizzaCost ( Size_Par, ToppingCount_Par, DeliveryRequested_Par ):
    basePrice = 0.0
    
    if Size_Par == "small":
        basePrice = 12.99
    elif Size_Par == "medium":
        basePrice = 15.99
    else:
        basePrice = 18.99
    
    toppingCost = ToppingCount_Par * TOPPING_PRICE
    
    deliveryFee = 0
    if DeliveryRequested_Par == True:
        deliveryFee = DELIVERY_FEE
    
    totalPrice = basePrice + toppingCost + deliveryFee
    
    print ( "Pizza order:" )
    print ( f"\tBase price: {basePrice:.2f}" )
    print ( f"\tTopping price: {toppingCost:.2f}" )
    print ( f"\tDelivery price: {deliveryFee:.2f}" )
    print ( f"\t\tTotal price: {totalPrice:.2f}" )

def main():
    printPizzaCost ( "small", 2, True )

if __name__ == "__main__":
    main()
